src/pages/LoginAndRegister/GuestAccountRedirect.vue

```diff
@@ -14,13 +14,28 @@ export default {
 						id
 						nonTrivialItemCount
 					}
+					my {
+						userAccount {
+							id
+						}
+					}
 				}`,
 			}).then(({ data }) => {
 				// Redirect to /checkout if there are items in the basket, otherwise /portfolio
+				const finalUrl = data?.shop?.nonTrivialItemCount > 0 ? '/checkout' : '/portfolio';
 				// Add claimed=1 to the url to show a confirmation tip message on the page
+				const query = { claimed: 1 };
+
+				let redirectPath = finalUrl;
+
+				// Check to see if user is authenticated
+				if (!data?.my?.userAccount?.id) {
+					redirectPath = '/ui-login';
+					query.doneUrl = finalUrl;
```

Comment:
I think you'll need to also set `query.login_hint = 'login';` so that the login page is redirected to instead of register.

Reply:
should we add the login_hint for both scenarios?